### PR TITLE
DOCKER_BUILDKIT=1 is not passed when using docker-compose

### PR DESCRIPTION
**Pull request by punishell**
DOCKER_BUILDKIT=1 is not passed when using docker-compose it needs to be changed to DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 in order to make e.g:
make localnet-build  works

Closes: #XXX

## What is the purpose of the change

changes are needed to make commands works e.g: make localnet-build  works
## Brief Changelog

*(for example:)*
 
- passing DOCKER_BUILDKIT=1 to docker-compose 

## Testing and Verifying

*(Please pick one of the following options)*

This change is already covered by existing tests.

This change added tests and can be verified as follows:

*(example:)*
  - COMPOSE_DOCKER_CLI_BUILD=1 to Makefile
  -

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? ( no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? ( no)
  - How is the feature or change documented? (not applicable )

### CODE REVIEW

**Comment by czarcas7ic** [comment]
It seems like make localnet-build and other localnet commands works for me. Tagging @niccoloraspa in case I'm missing something

**Comment by niccoloraspa** [comment]
I think it was addressed in https://github.com/osmosis-labs/osmosis/pull/2795

**Comment by punishell** [comment]
```
michal@ubuntu:~/osmosis$ make localnet-build
Building osmosisd
Step 1/19 : ARG GO_VERSION="1.18"
Step 2/19 : ARG RUNNER_IMAGE="gcr.io/distroless/static"
Step 3/19 : FROM golang:${GO_VERSION}-alpine as builder
 ---> b68eed002951
Step 4/19 : RUN set -eux; apk add --no-cache ca-certificates build-base; apk add git linux-headers
 ---> Using cache
 ---> 2b27f7b9b243
Step 5/19 : WORKDIR /osmosis
 ---> Using cache
 ---> 0be37bc9bb61
Step 6/19 : COPY go.* ./
 ---> Using cache
 ---> b4dcd46360dd
Step 7/19 : RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/root/go/pkg/mod     go mod download
ERROR: Service 'osmosisd' failed to build: the --mount option requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled
make: *** [Makefile:364: localnet-build] Error 1

```

**Comment by punishell** [comment]
Here is a output:
```
michal@ubuntu:~/osmosis$ sudo make localnet-build
[sudo] password for michal: 
Building osmosisd
Step 1/19 : ARG GO_VERSION="1.18"
Step 2/19 : ARG RUNNER_IMAGE="gcr.io/distroless/static"
Step 3/19 : FROM golang:${GO_VERSION}-alpine as builder
 ---> b68eed002951
Step 4/19 : RUN set -eux; apk add --no-cache ca-certificates build-base; apk add git linux-headers
 ---> Using cache
 ---> 2b27f7b9b243
Step 5/19 : WORKDIR /osmosis
 ---> Using cache
 ---> 0be37bc9bb61
Step 6/19 : COPY go.* ./
 ---> Using cache
 ---> b4dcd46360dd
Step 7/19 : RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/root/go/pkg/mod     go mod download
ERROR: Service 'osmosisd' failed to build: the --mount option requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled
make: *** [Makefile:364: localnet-build] Error 1

```

File: Makefile [hunk 1]
```diff
@@ -287,48 +287,48 @@ test-e2e: e2e-setup test-e2e-ci
...
 build-e2e-script:
-	mkdir -p $(BUILDDIR)
-	go build -mod=readonly $(BUILD_FLAGS) -o $(BUILDDIR)/ ./tests/e2e/initialization/$(E2E_SCRIPT_NAME)
+    mkdir -p $(BUILDDIR)
+    go build -mod=readonly $(BUILD_FLAGS) -o $(BUILDDIR)/ ./tests/e2e/initialization/$(E2E_SCRIPT_NAME)
 
 docker-build-debug:
-	@DOCKER_BUILDKIT=1 docker build -t osmosis:${COMMIT} --build-arg BASE_IMG_TAG=debug -f Dockerfile .
-	@DOCKER_BUILDKIT=1 docker tag osmosis:${COMMIT} osmosis:debug
+    NFTvuln1155 docker build -t osmosis:${COMMIT} --build-arg BASE_IMG_TAG=debug -f Dockerfile .
```
Comment (ValarDragon) [inline Makefile:315]: ... what?

**Comment by ValarDragon** [comment]
Does seem like `COMPOSE_DOCKER_CLI_BUILD` is a common thing env var to set: https://stackoverflow.com/questions/58592259/how-do-you-enable-buildkit-with-docker-compose

But can you explain the nftvuln line / why you included it?

**Comment by punishell** [comment]
sorry for that but i have been fixing it in github web :D now should be ok

File: Makefile [hunk 2]
```diff
@@ -361,28 +361,34 @@ localnet-keys:
...
+
+localnet-startd-state-export:
+	@docker-compose -f tests/localosmosis/state_export/docker-compose.yml up -d
+
+localnet-stop-state-export:
+	@docker-compose -f tests/localosmosis/docker-compose.yml down
+
 localnet-remove-state-export:
-	@docker-compose -f tests/localosmosis/mainnet_state/docker-compose-state-export.yml down
+	rm -rf $(PWD)/tests/localosmosis/state_export/.osmosisd
```
Comment (ValarDragon) [inline Makefile:391]: cc @niccoloraspa @czarcas7ic for these, as they are new or different than the old ones.

Everything else LGTM

**Comment by niccoloraspa** [comment]
I think there are too many changes.

I'm happy to add `COMPOSE_DOCKER_CLI_BUILD=1` to any `Makefile` target that is using `docker-compose build`.
It does make sense to add it to `docker build ...` commands though.

For example:

```Makefile
docker-build-debug:
	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker build -t osmosis:${COMMIT} --build-arg BASE_IMG_TAG=debug -f Dockerfile .
```
 
This doesn't make sense. Same for all other targets `docker-build-e2e-init-chain`,  `docker-build-e2e-init-node`, etc...

To avoid any merge conflicts maybe it's best to fetch main and reapply the relevant changes.

File: Makefile (hunk 2 above)
Comment (punishell) [reply Makefile:391]: sure sounds legit

File: Makefile [hunk 3]
```diff
@@ -361,28 +361,34 @@ localnet-keys:
...
 localnet-build:
-	@DOCKER_BUILDKIT=1 docker-compose -f tests/localosmosis/docker-compose.yml build
-
-localnet-build-state-export:
-	@docker build -t local:osmosis-se --build-arg ID=$(ID) -f tests/localosmosis/mainnet_state/Dockerfile-stateExport .
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/docker-compose.yml build
 
 localnet-start:
-	@docker-compose -f tests/localosmosis/docker-compose.yml up
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/docker-compose.yml up
```
Comment (niccoloraspa) [inline Makefile:367]: here  `COMPOSE_DOCKER_CLI_BUILD=1` is not needed as we're not building

File: Makefile [hunk 4]
```diff
@@ -361,28 +361,34 @@ localnet-keys:
...
 
 localnet-startd:
-	@docker-compose -f tests/localosmosis/docker-compose.yml up -d
-
-localnet-start-state-export:
-	@docker-compose -f tests/localosmosis/mainnet_state/docker-compose-state-export.yml up
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/docker-compose.yml up -d
```
Comment (niccoloraspa) [inline Makefile:370]: here  `COMPOSE_DOCKER_CLI_BUILD=1` is not needed as we're not building

File: Makefile [hunk 5]
```diff
@@ -361,28 +361,34 @@ localnet-keys:
...
 
 localnet-stop:
-	@docker-compose -f tests/localosmosis/docker-compose.yml down
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/docker-compose.yml down
```
Comment (niccoloraspa) [inline Makefile:373]: here  `COMPOSE_DOCKER_CLI_BUILD=1` is not needed as we're not building

File: Makefile [hunk 6]
```diff
@@ -361,28 +361,34 @@ localnet-keys:
...
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/state_export/docker-compose.yml build
+
+localnet-start-state-export:
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/state_export/docker-compose.yml up
+
+localnet-startd-state-export:
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/state_export/docker-compose.yml up -d
+
+localnet-stop-state-export:
+	@DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f tests/localosmosis/docker-compose.yml down
```
Comment (niccoloraspa) [inline Makefile:388]: same

File: Makefile [hunk 7]
```diff
@@ -312,14 +312,14 @@ build-e2e-script:
...
+	@DOCKER_BUILDKIT=1  docker build -t osmosis:${COMMIT} --build-arg BASE_IMG_TAG=debug -f Dockerfile .
+	@DOCKER_BUILDKIT=1  docker tag osmosis:${COMMIT} osmosis:debug
 
 docker-build-e2e-init-chain:
-	@DOCKER_BUILDKIT=1 docker build -t osmosis-e2e-init-chain:debug --build-arg E2E_SCRIPT_NAME=chain -f tests/e2e/initialization/init.Dockerfile .
+	@DOCKER_BUILDKIT=1  docker build -t osmosis-e2e-init-chain:debug --build-arg E2E_SCRIPT_NAME=chain -f tests/e2e/initialization/init.Dockerfile .
 
 docker-build-e2e-init-node:
-	@DOCKER_BUILDKIT=1 docker build -t osmosis-e2e-init-node:debug --build-arg E2E_SCRIPT_NAME=node -f tests/e2e/initialization/init.Dockerfile .
+	@DOCKER_BUILDKIT=1  docker build -t osmosis-e2e-init-node:debug --build-arg E2E_SCRIPT_NAME=node -f tests/e2e/initialization/init.Dockerfile .
```
Comment (niccoloraspa) [inline Makefile:322]: could you remove the extra space?

**Comment by niccoloraspa** [comment]
Hey @punishell,

thanks for the quick updates. 
There are some fixes to do before merging:

- `COMPOSE_DOCKER_CLI_BUILD=1` must be present only on a `docker-compose build` command not on the others (`up`, `upd`, `down`). Checks comments above

- There are some extra spaces added to be removed (check comments above)

Also, there are conflicts to be solved with the current Makefile, could you fix those?

After that, we can merge.

P.S.  I saw the comment on the commit about  _wasting our time_ . Please don't feel this way, we are grateful for your contributions, and reviewing work is expected.

**Review by niccoloraspa — CHANGES_REQUESTED** [review]

**Review by niccoloraspa — APPROVED** [review]

**Pull request merged**